packages/sit-onyx/src/components/OnyxTimer/OnyxTimer.vue

```diff
@@ -0,0 +1,68 @@
+<script lang="ts" setup>
+import { computed, watchEffect } from "vue";
+import { useTimer } from "../../composables/useTimer";
+import { injectI18n } from "../../i18n";
+import { formatTime, timeToDurationString } from "../../utils/time";
+import type { OnyxTimerProps } from "./types";
+
+const props = defineProps<OnyxTimerProps>();
+
+const emit = defineEmits<{
+  /**
+   * Emitted when timer has ended.
+   */
+  timerEnded: [];
+}>();
+
+const { locale } = injectI18n();
+const { timeLeft, isEnded } = useTimer(computed(() => props.endTime));
+
+const timeFormat = computed(
+  () => new Intl.RelativeTimeFormat(locale.value, { numeric: "always", style: "short" }),
+);
+
+/**
+ * Formatted remaining time.
+ */
+const formattedTime = computed(() => formatTime(timeLeft.value, timeFormat.value));
+
+watchEffect(() => isEnded.value && emit("timerEnded"));
+</script>
+
+<template>
+  <div
+    class="onyx-timer onyx-text"
+    role="timer"
+    :aria-label="props.hideLabel ? props.label : undefined"
```

Comment:
Since the label is required, why don't we always set :aria-label?
We are doing it in the button.

Reply:
In the other form component like input, switch etc. we also do it like this only add it when hidedLabel is true